hazelcast/include/hazelcast/client/imap.h

```diff
@@ -101,7 +101,8 @@ class HAZELCAST_API imap : public proxy::IMapImpl
     template<typename K, typename V>
     boost::future<boost::optional<V>> get(const K& key)
     {
-        return to_object<V>(get_internal(to_data(key)));
+        return controlled_serialization(
+          [&]() { return to_object<V>(get_internal(to_data(key))); });
```

Comment:
Reference capture of key. This may create a problem if user provided key is deleted? Please also use explicit capture of each variable rather than default capture, for better visibility.

Reply:
I believe it should be ok since we will use key only in this thread. It will not be delegated to another thread.